packages/react-components/src/templates/TagsPageBody.tsx

```diff
@@ -56,11 +66,25 @@ const TagsPageBody: React.FC<TagsPageBodyProps> = ({
       />
     }
   >
-    {results.map((data) => (
-      <div key={data.id}>
-        <SharedResearchCard {...data} />
-      </div>
-    ))}
+    {results.map((data) => {
+      if ('documentType' in data) {
```

Comment:
I'm not sure if I understood the suggestion. I thought about doing something like

```
const tagFieldByEntity = {
    'research-output': 'keywords',
    'user': 'expertiseAndResourceTags'
  }
  
  const EntityCard: React.FC<TagsPageBodyProps['results'][number]> = ({
    ...data
  }) => {
    if (tagFieldByEntity['research-output'] in data) {
      return <SharedResearchCard {...data} />;
    }
  
    if (tagFieldByEntity['user'] in data) {
      return <PeopleCard {...data} />;
    }
  
    return null;
  };
```

But then TS give me an error because it can't make the type narrowing properly as `tagFieldByEntity['research-output']` is recognized as a string not `keywords` exactly

<img width="592" alt="Screenshot 2023-10-30 at 07 21 54" src="https://github.com/yldio/asap-hub/assets/16595804/76584ce3-ef60-464a-82a1-a86708885ac5">